compiler/passes/src/type_checking/check_expressions.rs

```diff
@@ -875,7 +875,16 @@ impl ExpressionVisitor for TypeChecker<'_> {
                 parse_integer_literal::<i128>(self.handler, string, input.span(), "i128");
                 Type::Integer(IntegerType::I128)
             }
-            Literal::Group(..) => Type::Group,
+            Literal::Group(s, ..) => {
+                let mut s = s.trim_start_matches('0');
+                if s.is_empty() {
+                    s = "0";
+                }
+                if format!("{s}group").parse::<snarkvm::prelude::Group<snarkvm::prelude::TestnetV0>>().is_err() {
+                    self.emit_err(TypeCheckerError::invalid_int_value(s, "group", input.span()));
+                }
+                Type::Group
+            }
             Literal::Scalar(..) => Type::Scalar,
```

Comment:
Does this case and the `Field` case need to be handled?

Reply:
I think the reason we need this for Group but not for Field or Scalar is because some decimals that are in the right range (i.e., field elements) are not valid x coordinates, like `1group`.  But I'm not sure where the range checks on Field and Scalar are.